packages/core/src/router/AbstractRouter.js

```diff
@@ -323,33 +323,38 @@ export default class AbstractRouter extends Router {
       return Promise.reject(error);
     }
 
+    const originalPath = this._getCurrentlyRoutedPath();
+    const { route } = this._getRouteHandlersByPath(originalPath);
```

Comment:
Method `_getRouteHandlersByPath` can return route property as undefined. It will crash for `route.extractParameters`.

Reply:
Fixed and also moved to dedicated method so it's not duplicated in `handleNotFound` method.